lib/str.ts

```diff
@@ -994,6 +994,7 @@ const Str = {
      * without query parameters
      */
     getExtension(url: string): string | undefined {
+        if (typeof url !== 'string') return undefined;
```

Comment:
Based on [this suggestion](https://github.com/Expensify/App/issues/46117#issuecomment-2326532872), let's also  add a `Log.warn` in here or at least a `console.warn`?

Reply:
Log please, adding conosle is useless as we don't have access to people's consoles.

Reply:
So I tried adding the `Log.warn` or `Log.info` but there were no logs in Expensify Web App console. There's this line in `Log.jsx` in `expensify-common` which has `window.DEBUG`:

https://github.com/Expensify/expensify-common/blob/42663c3639564a18da9391e7b0590768e467e3ba/lib/Log.jsx#L38

This `window.DEBUG` is undefined in the web app console when printed from `node_modules/expensify-common/Log.js`.

In Expensify codebase, we have explicitly marked `isDebug: true` [here](https://github.com/Expensify/App/blob/fdc05854f0fa97bb9744deff2e0b1a7616636b1a/src/libs/Log.ts#L81).

When I set `isDebug: true` in `expensify-common/Log.jsx`, I start getting logs in Expensify Web App as well.

Does anyone know if there's something I am missing with `window.DEBUG` while testing?

<hr/>

Once the above succeeds, I plan to do something like below so we have logs in crashlytics as well, this is just an example:

```diff
diff --git a/src/libs/Log.ts b/src/libs/Log.ts
index 72673b8d3f..eb29b3ac08 100644
--- a/src/libs/Log.ts
+++ b/src/libs/Log.ts
@@ -13,6 +13,8 @@ import {shouldAttachLog} from './Console';
 import getPlatform from './getPlatform';
 import * as Network from './Network';
 import requireParameters from './requireParameters';

 let timeout: NodeJS.Timeout;
 let shouldCollectLogs = false;
@@ -73,6 +75,12 @@ const Log = new Logger({
 
         flushAllLogsOnAppLaunch().then(() => {
             console.debug(message, extraData);
+
+            if (message.indexOf('Str.getExtension') !== -1) {
+                crashlytics().log(message);
+                crashlytics().recordError(new Error(message, {cause: JSON.stringify(extraData)}));
+            }
+
             if (shouldCollectLogs) {
                 addLog({time: new Date(), level: CONST.DEBUG_CONSOLE.LEVELS.DEBUG, message, extraData});
             }
```

Let me know what you think of it 👍

Reply:
Both `Log.warn` and `Log.info` should log in the console too... do you have all log levels enabled in the JS console?
<img width="123" alt="image" src="https://github.com/user-attachments/assets/8a863249-2b03-42d4-8a9f-e19866a8f3c2">

Reply:
@iwiznia Yes I do have this enabled.

The issue is that `window.DEBUG` is `undefined` and as a result of which `isDebug` is set to `false` and we don't get any logs. Below is the diff from `node_modules/expensify-common`:

![Screenshot 2024-09-06 at 1 55 50 PM](https://github.com/user-attachments/assets/081c964f-7ef7-4468-a474-9a8a485cf2f5)


<details>

<summary>See diff</summary>

```diff
diff --git a/node_modules/expensify-common/dist/Log.js b/node_modules/expensify-common/dist/Log.js
index b7ffa9f..1202637 100644
--- a/node_modules/expensify-common/dist/Log.js
+++ b/node_modules/expensify-common/dist/Log.js
@@ -56,6 +56,8 @@ function clientLoggingCallback(message) {
         console.log(message);
     }
 }
+
+console.log('=== window.DEBUG ', window.DEBUG);
 exports.default = new Logger_1.default({
     serverLoggingCallback,
     clientLoggingCallback,
diff --git a/node_modules/expensify-common/dist/str.js b/node_modules/expensify-common/dist/str.js
index 2ebc6eb..3b2d83e 100644
--- a/node_modules/expensify-common/dist/str.js
+++ b/node_modules/expensify-common/dist/str.js
@@ -23,6 +23,9 @@ var __importStar = (this && this.__importStar) || function (mod) {
     __setModuleDefault(result, mod);
     return result;
 };
+var __importDefault = (this && this.__importDefault) || function (mod) {
+    return (mod && mod.__esModule) ? mod : { "default": mod };
+};
 Object.defineProperty(exports, "__esModule", { value: true });
 /* eslint-disable no-control-regex */
 const awesome_phonenumber_1 = require("awesome-phonenumber");
@@ -30,6 +33,7 @@ const HtmlEntities = __importStar(require("html-entities"));
 const Constants = __importStar(require("./CONST"));
 const UrlPatterns = __importStar(require("./Url"));
 const Utils = __importStar(require("./utils"));
+const Log_1 = __importDefault(require("./Log"));
 const REMOVE_SMS_DOMAIN_PATTERN = /@expensify\.sms/gi;
 function resultFn(parameter, ...args) {
     if (typeof parameter === 'function') {
@@ -909,6 +913,10 @@ const Str = {
      */
     getExtension(url) {
         var _a, _b;
+        if (typeof url !== 'string') {
+            Log_1.default.warn('=== Str.getExtension: url is not a string', { url });
+            return undefined;
+        }
         return (_b = (_a = url.split('.').pop()) === null || _a === void 0 ? void 0 : _a.split('?')[0]) === null || _b === void 0 ? void 0 : _b.toLowerCase();
     },
     /**
```

</details>

When I set `isDebug: true` directly in `node_modules/expensify-common/dist/Log.js`, I start to get logs. Below is the update:

```js

//node_modules/expensify-common/dist/Log.js

console.log('=== window.DEBUG ', window.DEBUG);
exports.default = new Logger_1.default({
    serverLoggingCallback,
    clientLoggingCallback,
    isDebug: true// Utils.isWindowAvailable() ? window.DEBUG : false,
});

```

Log is visible in the console when `isDebug: true`:

![Screenshot 2024-09-06 at 1 56 26 PM](https://github.com/user-attachments/assets/e3623dc3-4899-4670-90cc-0cd201d7f0f0)

<hr/>

I don't have any context on `window.DEBUG`, so it will be great if someone can share details on it and why it can be `undefined`.

Reply:
I don't know how that debug flag is working, but it will work without any changes in production.

Reply:
Oh.. So how was it tested when `window.DEBUG` was first introduced?

Anyways, I will update the PR with `Log.warn` and once this is merged, I will raise a PR to Expensify/App to incorporate this log in crashlytics.

Reply:
> Oh.. So how was it tested when window.DEBUG was first introduced?

I don't know 🤷 

> Anyways, I will update the PR with Log.warn and once this is merged, I will raise a PR to Expensify/App to incorporate this log in crashlytics.

Do we need it in crashalytics? `Log.warn` should already make it go to our logs.

Reply:
@francoisl So using `Log.alert()` appears to not be that helpful because it lacks the symbolication for the stack trace. I have recorded this by calling `getExtension` with numeric data in `App.tsx` while setting webpack to `production` mode in `webpack.dev.ts`.

Do you know it will show appropriate function names when used in actual prod?


![Screenshot 2024-09-09 at 1 05 35 PM](https://github.com/user-attachments/assets/4502e766-22b0-451a-b69a-274b8d434618)

Reply:
Is that stack trace useful? It says webpack-internal in all of them....

Reply:
Oh ok I see, it doesn't look useful indeed. Sounds good for `Log.warn` then.

Reply:
My guess is that it won't make it easier for us to find where this is being called wrong.... but we'll see.